### PR TITLE
feat(range): add neutral point

### PR DESCRIPTION
**Pull request by KillerCodeMonkey**
#### Short description of what this resolves:
Ability to set the neutral point of the picker. Active ticks and bar are measured from the neutral point instead of the min value.

bildschirmfoto 2019-02-06 um 13 38 18

#### Changes proposed in this pull request:

- added neutralPoint prop
- calculate active state of ticks and bar
- set intial value to neutralPoint if not set

**Ionic Version**:
4.0.0

### CODE REVIEW

**Comment by abennouna** [comment]
Would this work in RTL as well?

**Comment by KillerCodeMonkey** [comment]
@abennouna i nothing changed in the base calculations. Maybe you can test this on your https://github.com/ionic-team/ionic/pull/17384 ?

**Comment by abennouna** [comment]
@KillerCodeMonkey I guess I'll wait till #17384 is merged then test yours if it's approved.

**Comment by KillerCodeMonkey** [comment]
@abennouna okay :)

File: core/src/components/range/range.tsx [hunk 1]
```diff
@@ -40,6 +40,24 @@ export class Range implements ComponentInterface {
    */
   @Prop() mode!: Mode;
 
+  /**
+   * The neutral point of the range slider.
+   * Default: value is `0` or the `min` when `neutralPoint < min` or `max` when `max < neutralPoint`.
+   */
+  @Prop() neutralPoint = 0;
+  protected neutralPointChanged() {
+    if (!this.noUpdate) {
```
Comment (liamdebeasi) [inline core/src/components/range/range.tsx]: I typically prefer to check the positive case when possible. It makes the code a bit more cleaner/readable and avoids extra levels of indentation.

In this case, you'd do `if (this.noUpdate) { return }`

Comment (KillerCodeMonkey) [reply core/src/components/range/range.tsx]: you are absolutly right, but this is done everywhere in the component, where `noUpdate` is used, so i wanted to make less changes as possible to the original code, but keep the "form" how the old code is structured.

But i will change it.

Thanks for looking into it :)

Comment (liamdebeasi) [reply core/src/components/range/range.tsx]: Fair point! This is something we should probably change elsewhere as well. Thanks for bringing this to our attention 😄

Comment (KillerCodeMonkey) [reply core/src/components/range/range.tsx]: should be solved 🙃

File: core/src/components/range/readme.md [hunk 2]
```diff
@@ -76,30 +84,39 @@ left or right of the range.
...
+| `mode`         | `mode`          | The mode determines which platform styles to use.                                                                                                                                                                                                                      | `"ios" \| "md"`                                       | `undefined` |
+| `name`         | `name`          | The name of the control, which is submitted with the form data.                                                                                                                                                                                                        | `string`                                              | `''`        |
+| `neutralPoint` | `neutral-point` | The neutral point of the range slider. Default: value is `0` or the `min` when `neutralPoint < min` or `max` when `max < neutralPoint`.                                                                                                                                | `number`                                              | `0`         |
```
Comment (liamdebeasi) [inline core/src/components/range/readme.md]: So in the documentation here you say the neutral point will be `min` when `neutralPoint < min` but the case you check  is `min < neutralPoint` (https://github.com/ionic-team/ionic/pull/17400/files#diff-96cc2845be9d5c7024e04f2e1856b65fR57). Which is the correct one?

Comment (KillerCodeMonkey) [reply core/src/components/range/readme.md]: nice catch 👍

**Comment by liamdebeasi** [comment]
Looks great! If you can just fix the merge conflicts, we should be good to merge! 👍

**Comment by KillerCodeMonkey** [comment]
@liamdebeasi merge conflicts resolved and fixed the active bar styling :)

**Comment by liamdebeasi** [comment]
@KillerCodeMonkey Looks like something got changed during your merge. The tick marks on the ranges are no longer visible.

This is what `master` looks like: 
screen shot 2019-02-12 at 4 35 23 pm

This is what your branch looks like:
screen shot 2019-02-12 at 4 35 50 pm

**Comment by KillerCodeMonkey** [comment]
@liamdebeasi sry my fault. The any types are getting me crazy. Maybe this would be another part of improvement 😉

**Comment by abennouna** [comment]
This actually causes a regression in RTL mode.

**Comment by liamdebeasi** [comment]
Hi @abennouna,

Would you be able to file a bug report?

Thanks!

**Comment by abennouna** [comment]
Maybe we also need RTL tests?

**Comment by liamdebeasi** [comment]
That would be good to have. I can discuss it with the team.

**Comment by KillerCodeMonkey** [comment]
@abennouna if you let me know, what you expect in the behaviour of rtl in this case i will look into it ;), but only saying it is a "regression" is a little bit unclear.

**Comment by abennouna** [comment]
@KillerCodeMonkey the PR seems to cancel the fix from #17384

A screencast shows the difference:

**Before #17400:**
https://drive.google.com/open?id=1YZnVjL9eQXjkY_gT9EI3Vm1LDqMsacRC

**With PR #17400:**
https://drive.google.com/open?id=1C_wiZrqT-svopCjCC9f2Kd6EhVjV6Y2O

By reverting the PR, range work correctly again.

That's what I mean by regression.

**Comment by KillerCodeMonkey** [comment]
@abennouna i am working on it :)

**Comment by liamdebeasi** [comment]
Hey there! 

After some discussion with the team, we have decided to revert your changes from this PR. We really appreciate all of the time and effort you put into creating these updates for Ionic, but we are not certain this is something we would like to introduce to the code base. Specifically, we are not certain about the use cases for this feature. The native input, material design, and polymer specs do not have this neutral point feature. 

Would you be able to share with us the use case(s) you had in mind for this feature? After that the team can have another discussion about this PR.

Sorry for the confusion, and please don't hesitate to reach out if you have any questions! 🙂

**Comment by KillerCodeMonkey** [comment]
We have ranges from negative to positive values.

But the default is 0.

So If you Change the value you want to Show the User how this value changes
from the default. And often the Base/Default is Not the min value.

We are working in the health care and There we have such use cases quite
often, e.g. by doing diagnosis are define therapies.






Liam DeBeasi <notifications@github.com> schrieb am Mi., 20. Feb. 2019,
21:42:

**Comment by KillerCodeMonkey** [comment]
@liamdebeasi are there hopes for a "remerge" or do you discussed it again?

**Comment by KillerCodeMonkey** [comment]
@liamdebeasi @brandyscarney bump :)

**Comment by liamdebeasi** [comment]
Hey there! Sorry for the delay - I thought I had replied to this already. (I guess not 🙂)

At this time we aren't going to add this feature to the framework. Sorry for any confusion!

**Pull request merged**